packages/@lwc/engine-core/src/framework/base-bridge-element.ts

```diff
@@ -148,6 +148,12 @@ export function HTMLBridgeElementFactory(
     descriptors.attributeChangedCallback = {
         value: createAttributeChangedCallback(attributeToPropMap, superAttributeChangedCallback),
     };
+
+    // To avoid leaking private component details, accessing internals from outside a component is not allowed.
+    descriptors.attachInternals = {
+        value: undefined,
+    };
```

Comment:
Can we make this throw a helpful error message in dev mode?

Reply:
@nolanlawson added an [error message](https://github.com/salesforce/lwc/pull/3670/files#diff-2a802371ad4d01928bd675d69bea9c719ecc10ce7a0ece771b220bad7f61382fR157), let me know if the wording works.

Reply:
I'd say `attachInternals cannot be accessed outside of a component. Use this.attachInternals instead.` (I.e. tell them what to do to resolve the issue 🙂  )

Reply:
Good idea!